### ISSUE TITLE
Update hint text guidance so users know when to punctuate

### ISSUE THREAD

**Issue by EoinShaughnessy** [alphagov/govuk-design-system#2090]
## What

Update our [hint text guidance](https://design-system.service.gov.uk/components/text-input/#hint-text) so users know when to add punctuation.

## Why

Users have asked us to provide guidance on:

- [when to include or omit a full stop in hint text](https://ukgovernmentdigital.slack.com/archives/C6DMEH5R6/p1645710637156809)
- [whether examples in hint text should include quote marks](https://ukgovernmentdigital.slack.com/archives/C6DMEH5R6/p1645721777420959?thread_ts=1645710637.156809&cid=C6DMEH5R6)

Some more feedback:

> I tend to follow the rule that omits full stops in headings, and as hint text is prose/body copy, include the full stop. I think the Design System website is inconsistent across its examples, which doesn’t help.

> We at DWP have just published new hint text guidance which recommends including the full stop in all cases: https://design-system.dwp.gov.uk/components/hint-text
> In coming to this recommendation we did quite a lot of research on current practice with full stops, which is written up here: https://github.com/dwp/design-system-community-backlog/discussions/81

> I like the idea of having a whole page dedicated to guidance on Hint text. I sometimes struggle to find the hint text guidance (eg on not including links) on the GOV.UK Design System website.

> A lot of designers assume the guidance is there in the [GOV.UK](http://gov.uk/) design system but in fact it’s spread out in several different places under other components, and as people have pointed out the examples vary in how they do content.

## Who needs to work on this

Content designer

## Done when

- [ ] guidance updated

**Comment by EoinShaughnessy** [comment]
Related issue: https://github.com/alphagov/govuk-design-system/issues/1327

**Comment by EoinShaughnessy** [comment]
This issue covers [recommendation #9](https://docs.google.com/presentation/d/1TxZ17AJDMKI6djT1lEQj6wuH6U9edjZyRm2MfmGOar8/edit#slide=id.g119f135a53b_1_5) in @SaraC19's support analysis slides.

**Comment by matteason** [comment]
It would be great to get some updated guidance on this - it's just come up on a project I'm working on and we found inconsistencies in examples, even for the same component. I'm erring on the side of no full stop because it's more consistent with labels and error messages but there's a fairly even split between the examples on the DS

![A list of checkboxes with the hint 'Select all that apply', with a full stop at the end](https://github.com/alphagov/govuk-design-system/assets/1935173/19d152e0-218a-44b4-a6bd-1a83279096bf)

![A list of checkboxes with the hint 'Select all countries that apply', with no full stop at the end](https://github.com/alphagov/govuk-design-system/assets/1935173/bd06c46d-d927-4d77-ba11-f5d91f51b416)

**Comment by calvin-lau-sig7** [comment]
In October 2024, we worked to tidy up instances of hint text on the pages and examples of the GOV.UK Design System to not include full stops. 

See: https://github.com/alphagov/govuk-design-system/pull/4169

We've done this to:
- keep to the original intention for hint text in the Design System
- stay consistent with similar text elements like input field labels and error messages
- stay consistent with the [guidance and example shown in the Service Manual](https://www.gov.uk/service-manual/design/writing-for-user-interfaces)
- encourage services to write hints as a single short simple sentence

We've also added guidance on relevant pages to make this clearer, along with some context on how users of assistive technologies perceive hint text.

We know there's been inconsistency on this and are aware of similar work, such as [DWP Design System exploring hint text in 2021](https://design-system.dwp.gov.uk/components/hint-text#end-with-a-full-stop), and would invite further feedback and discussion.
